lightningd/peer_control.c

```diff
@@ -1200,10 +1203,39 @@ static struct command_result *json_setchannelfee(struct command *cmd,
 		   NULL))
 		return command_param_failed();
 
-	// TODO: write this stuff
+	// find the channel
+	peer = peer_from_json(cmd->ld, buffer, idtok);
+	if (peer) {
+		channel = peer_active_channel(peer);
+		if (!channel)
+			return command_fail(cmd, LIGHTNINGD,
+				    "Could not find active channel of peer with that id");
+	} else {
+		struct command_result *res;
+		res = command_find_channel(cmd, buffer, idtok, &channel);
+		if (res)
+			return res;
+	}
+	if (!channel)
+		return command_fail(cmd, LIGHTNINGD, "Could not find peer with that id");
+
+	// set actual values
+	channel->feerate_base = *base;
+	channel->feerate_ppm = *ppm;
+	// TODO: call channel save manually?
+
+	struct channel_id cid;
```

Comment:
Variable declarations should be at the top of the function.

Reply:
sounds legit, will be fixed